### PR TITLE
Load configuration from property file and performance fixes

### PR DESCRIPTION
**Pull request by SlavoKrupa**
* the property file with configuration can be specified as -p FILE
* SilverWare now ends with zero code when everything is done
* Add timeout for shutdown thread
* Port can be stored as string in properties now
* Just single **local** lookup per rest service now occurs

### CODE REVIEW

File: microservices/src/main/java/io/silverware/microservices/Boot.java [hunk 1]
```diff
@@ -47,6 +47,7 @@
 public final class Boot {
 
    private static final Logger log = LogManager.getLogger(Boot.class);
+   static boolean IS_TESTING = false;
```
Comment (livthomas) [inline microservices/src/main/java/io/silverware/microservices/Boot.java]: I do not think this is a good solution. What about using a system property instead? Something like `silverware.system.exit` which would be set to `true` by default.

File: microservices/src/main/java/io/silverware/microservices/Boot.java [hunk 2]
```diff
@@ -89,27 +95,18 @@ private static void logFlush() {
...
-      try {
-         props.load(Boot.class.getResourceAsStream("silverware.properties"));
-      } catch (NullPointerException | IOException ioe) {
-         log.info("No configuration property file available. Using default values.");
-      }
-
-      return props;
-   }
 
    private static Properties loadProperties(final File propertiesFile) {
```
Comment (livthomas) [inline microservices/src/main/java/io/silverware/microservices/Boot.java:104]: `propertiesFile` parameter is not mentioned in the JavaDoc.

File: microservices/src/main/java/io/silverware/microservices/Boot.java (hunk 1 above)
Comment (SlavoKrupa) [reply microservices/src/main/java/io/silverware/microservices/Boot.java]: Done. I used existing property `Executor.SHUTDOWN_HOOK`.

File: microservices/src/main/java/io/silverware/microservices/Boot.java (hunk 2 above)
Comment (SlavoKrupa) [reply microservices/src/main/java/io/silverware/microservices/Boot.java:104]: added to javadoc.

**Pull request merged**